packages/@webex/plugin-meetings/src/annotation/index.ts

```diff
@@ -0,0 +1,316 @@
+import uuid from 'uuid';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import {TriggerProxy as Trigger} from '@webex/plugin-meetings';
+import {WebexPlugin, config} from '@webex/webex-core';
+
+import {
+  EVENT_TRIGGERS,
+  ANNOTATION_RELAY_TYPES,
+  ANNOTATION,
+  ANNOTATION_REQUEST_TYPE,
+  ANNOTATION_ACTION_TYPE,
+  ANNOTATION_RESOURCE_TYPE,
+} from './constants';
+
+import {StrokeData, RequestData, IAnnotationChannel, CommandRequestBody} from './annotation.types';
+import {HTTP_VERBS} from '../constants';
+
+/**
+ * @description Annotation to handle LLM and Mercury message and locus API
+ * @class
+ */
+class AnnotationChannel extends WebexPlugin implements IAnnotationChannel {
+  namespace = ANNOTATION;
+
+  private seqNum: number;
+
+  hasSubscribedToEvents: boolean;
+
+  approvalUrl: string;
+  locusUrl: string;
+  annotationStatus: string;
+  deviceUrl: string;
+
+  /**
+   * Initializes annotation module
+   */
+  constructor(...args) {
+    super(...args);
+    this.seqNum = 1;
+  }
+
+  /**
+   * Process Stroke Data
+   * @param {request} request stroke data
+   * @returns {void}
+   */
+  private processStrokeMessage(request) {
+    this.decryptContent(request.encryptionKeyUrl, request.content).then((decryptedContent) => {
+      request.content = decryptedContent;
+      Trigger.trigger(
+        this,
+        {
+          file: 'annotation',
+          function: 'processStrokeMessage',
+        },
+        EVENT_TRIGGERS.ANNOTATION_STROKE_DATA,
+        {
+          data: request,
+        }
+      );
+    });
+  }
+
+  /** bind all events from mercury
+   * @param {Object} e
+   * @returns {undefined}
+   */
+  private eventCommandProcessor = (e) => {
+    this.seqNum = e.sequenceNumber + 1;
+
+    if (
+      e?.data?.eventType === 'locus.approval_request' &&
+      e?.data?.approval?.resourceType === ANNOTATION_RESOURCE_TYPE &&
+      e?.data?.approval?.actionType
+    ) {
+      Trigger.trigger(
+        this,
+        {
+          file: 'annotation',
+          function: 'approval_request',
+        },
+        EVENT_TRIGGERS.ANNOTATION_COMMAND,
+        {
+          type: e?.data?.approval?.actionType,
+          payload: e?.data?.approval,
+        }
+      );
+    }
+  };
+
+  /** bind all events from llm
+   * @param {Object} e
+   * @returns {undefined}
+   */
+  private eventDataProcessor = (e) => {
+    this.seqNum = e.sequenceNumber + 1;
+
+    switch (e?.data?.relayType) {
+      case ANNOTATION_RELAY_TYPES.ANNOTATION_CLIENT:
+        this.processStrokeMessage(e.data.request);
+        break;
+      default:
+        break;
+    }
+  };
+
+  /**
+   * Listen to websocket messages
+   * @returns {undefined}
+   */
+  private listenToEvents() {
+    if (!this.hasSubscribedToEvents) {
+      // @ts-ignore
+      this.webex.internal.mercury.on('event:locus', this.eventCommandProcessor);
```

Comment:
issue: I can see a test checking for llm listening, but I can't find a test checking for mercury listening

Reply:
Add mercury listening check